Source/Core/Common/WorkQueueThread.h

```diff
@@ -59,14 +59,13 @@ class WorkQueueThread
 
       while (true)
       {
-        T item;
-        {
-          std::unique_lock<std::mutex> lg(m_lock);
-          if (m_items.empty())
-            break;
-          item = m_items.front();
-          m_items.pop();
-        }
+        std::unique_lock lg(m_lock);
```

Comment:
You've changed the logic to hold this lock during the function call. Other than that the changes look good.

Reply:
Line 67 unlocks the mutex explicitly, so it's not held during the function call.

Reply:
Oops. Dunno how I missed that. I guess the removal of the `{}`s scared me. :P